protocols/autonat/src/protocol.rs

```diff
@@ -105,16 +105,14 @@ impl DialRequest {
         if msg.type_pb != Some(proto::MessageType::DIAL) {
             return Err(io::Error::new(io::ErrorKind::InvalidData, "invalid type"));
         }
-        let (peer_id, addrs) = if let Some(proto::Dial {
+
+        let Some(proto::Dial {
             peer:
                 Some(proto::PeerInfo {
                     id: Some(peer_id),
                     addrs,
-                }),
-        }) = msg.dial
-        {
-            (peer_id, addrs)
-        } else {
+                })
+        }) = msg.dial else {
             log::debug!("Received malformed dial message.");
```

Comment:
I think this log is kind of unnecessary when we already include all the info in the `io::Error`. Can you change this one to an `ok_or_else`?

Reply:
I tried `ok_or_else` 

```rust
let proto::Dial { peer: Some(proto::PeerInfo { id: Some(peer_id), addrs }) } = 
    msg.dial.ok_or_else(|| {
        log::debug!("Received malformed dial message.");
        io::Error::new(io::ErrorKind::InvalidData, "invalid dial message")
    })?;
```

but the let binding is a refutable pattern, i.e. has more than one variant

I came up with a solution using combinators, but I dont think it simplifies things here much

```rust
let peer_id_result = msg.dial.and_then(|dial| {
    dial.peer.and_then(|peer_info| {
        let Some(peer_id) = peer_info.id else { return None };
        Some((peer_id, peer_info.addrs))
    })
});

let (peer_id, addrs) = peer_id_result.ok_or_else(|| io::Error::new(io::ErrorKind::InvalidData, "invalid dial message"))?;

```

It does potentially make things a bit more readable. What do you think? I'll try to spend a bit more time here to see if I can simplify things here